source/auth/auth.md

```diff
@@ -1224,7 +1224,8 @@ in the MONGODB-OIDC specification, including sections or blocks that specificall
   - TOKEN_RESOURCE\
     The URI of the target resource. If `TOKEN_RESOURCE` is provided and `ENVIRONMENT` is not one of
     `["azure", "gcp"]` or `TOKEN_RESOURCE` is not provided and `ENVIRONMENT` is one of `["azure", "gcp"]`, the driver
-    MUST raise an error.
+    MUST raise an error. Drivers MUST ensure that `TOKEN_RESOURCE` is url-encoded, such as by using a regex for special
+    characters or by checking if unencoding the value changes it, and then encoding if necessary.
```

Comment:
I suspect the intent is to say "the value of TOKEN_RESOURCE must be url-DEcoded" (with no mention of checks; and we must then document that the value should be ENcoded by the user). 

If the intent is to tell drivers to, in tests, encode the environment variable before supplying it to tests, I think that the value should come pre-encoded out of any scripts that produce it.

A longer explanation to make sure I'm on the same page: supposing the following URLs start with `mongodb://localhost/?authMechanism=MONGODB-OIDC`, there should be no difference between these URLS:
```
&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:foo:bar

&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE%3Afoo%3Abar
```
That is, both should decode to an equivalent `authMechanismProperties` value, with no error. In Java, there was a check against a second `:` that I [removed](https://github.com/mongodb/mongo-java-driver/pull/1371#discussion_r1578189423), because it did not appear to violate any spec - it is easy enough to split on only the first `:`.

However, if we need to support other characters in TOKEN_RESOURCE, (for example, we need to allow `,`, `&`, `%`...), then we need to specify that the value of TOKEN_RESOURCE must be decoded by the driver (independently of the url-decoding already being done at the url params level). For example, the value `foo:bar,comma` would need to be encoded, and the following two would be equivalent:

```
&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:foo%3Abar%2Cbaz

&authMechanismProperties=ENVIRONMENT%3Aazure%2CTOKEN_RESOURCE%3Afoo%253Abar%252Cbaz
```
Note that there is a `:` in `TOKEN_RESOURCE:foo`, in contrast to the original two examples.